src/components/calcite-pagination/calcite-pagination.tsx

```diff
@@ -0,0 +1,219 @@
+  //
+  //  Private Properties
+  //
+  // --------------------------------------------------------------------------
+
+  @Element() el: HTMLElement;
+
+  @State() selectedIndex = this.num;
+  @Watch('selectedIndex') selectedIndexWatchHandler() {
+    this.calcitePaginationUpdate.emit({
+      start: this.start,
+      total: this.total,
+      num: this.selectedIndex
+    });
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  //--------------------------------------------------------------------------
+
+  /** Emitted whenever the selected page changes.
+   * @event calcitePaginationUpdate
+   */
+  @Event() calcitePaginationUpdate: EventEmitter;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Public Methods
+  //
+  // --------------------------------------------------------------------------
+
+  /** When called, selected page will increment by 1.
+   */
+  @Method()
+  async nextPage(): Promise<void> {
+    this.selectedIndex = Math.min(this.total, this.selectedIndex + 1);
+  }
+
+  /** When called, selected page will decrement by 1.
+   */
+  @Method()
+  async previousPage(): Promise<void> {
+    this.selectedIndex = Math.max(this.start, this.selectedIndex - 1);
+  }
+
+  /** Set selected page to a specific page number. Will not go below start or above total.
+   */
+  @Method()
+  async setPage(num: number): Promise<void> {
+    this.selectedIndex = Math.max(this.start, Math.min(this.total, num));
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  // --------------------------------------------------------------------------
+
+  previousClicked = (): void => {
+    this.previousPage();
+  };
+
+  nextClicked = (): void => {
+    this.nextPage();
+  };
+
+  showLeftEllipsis() {
+    return (this.selectedIndex - this.start) > 3;
+  }
+
+  showRightEllipsis() {
+    return (this.total - this.selectedIndex) > 3;
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Render Methods
+  //
+  //--------------------------------------------------------------------------
+
+  renderPages() {
+    let pages = [];
+    let currentNum;
+    let end;
+
+    if ( this.total <= maxPagesDisplayed ) {
+      currentNum = this.start + 1;
+      end = this.total - 1;
+    } else {
+      if ( this.selectedIndex < maxPagesDisplayed ) {
+        currentNum = this.start + 1;
+        end = this.start + 4;
+      } else {
+        if ( this.selectedIndex + 3 >= this.total ) {
+          currentNum = this.total - 4;
+          end = this.total -1;
+        } else {
+          currentNum = this.selectedIndex - 1;
+          end = this.selectedIndex + 1;
+        }
+      }
+    }
+
+    while (currentNum <= end) {
+      pages.push(currentNum);
+      currentNum ++;
+    }
+
+    return pages.map(page => {
```

Comment:
you could just online this as it has a single line return: `return pages.map(page => this.renderPage(page));`